File: ELEC881_Tabnine/src/No Description Prompt/Python/q6-ugly-number-ii.py
class Solution(object):
    def nthUglyNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        ugly = [0] * n
        ugly[0] = 1
        i2, i3, i5 = 0, 0, 0
        for i in range(1, n):
            ugly[i] = min(ugly[i2]*2, ugly[i3]*3, ugly[i5]*5)
            if ugly[i] == ugly[i2]*2: i2 += 1
            if ugly[i] == ugly[i3]*3: i3 += 1
            if ugly[i] == ugly[i5]*5: i5 += 1
        return ugly[n-1]
        

File: ELEC881_Tabnine/src/No Description Prompt/Java script (JS)/q6-ugly-number-ii.js
/**
 * @param {number} n
 * @return {number}
 */
var nthUglyNumber = function(n) {
    let ugly = new Array(n).fill(0);
    ugly[0] = 1;
    let i2 = 0, i3 = 0, i5 = 0;
    for (let i = 1; i < n; i++) {
        ugly[i] = Math.min(ugly[i2] * 2, ugly[i3] * 3, ugly[i5] * 5);
        if (ugly[i] === ugly[i2] * 2) i2++;
        if (ugly[i] === ugly[i3] * 3) i3++;
        if (ugly[i] === ugly[i5] * 5) i5++;
    }
    return ugly[n - 1];
};